S3T3n1/src/main/java/org/example/model/repository/ReadConnection.java

```diff
@@ -1,21 +1,98 @@
 package org.example.model.repository;
 
-import org.example.model.domain.Invoice;
-import org.example.model.domain.Product;
+import org.example.model.domain.*;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class ReadConnection implements Connector{
+    static List<Invoice> invoiceLogRetrieved = new ArrayList<>();
+    static List<Product> stockListRetrieved = new ArrayList<>();
+    static File invoiceLogDB = new File("InvoiceLogDB.txt");
```

Comment:
i would leave the declared fields up here, but the initialization within a try/catch block in the connect() method, which i also implemented in the boot() method within the app as it will need to do so automatically since its dealing with 1 database and not a given option. i think if you repeat my try catch block within the writeconnection connect() method it should work and not create any new files if it already exists.